WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -387,6 +380,9 @@ final class ProductFormViewController<ViewModel: ProductFormViewModelProtocol>:
                 }
                 eventLogger.logDescriptionTapped()
                 editProductDescription()
+            case .promoteWithBlaze:
+                ServiceLocator.analytics.track(event: .Blaze.blazeEntryPointTapped(source: .productDetailPromoteButton))
```

Comment:
We want to track this only if the intro view will not be displayed - so that we have the correct `source` for the Blaze entry point. Let's check for the intro view first:

```suggestion
                if !viewModel.shouldShowBlazeIntroView {
                    ServiceLocator.analytics.track(event: .Blaze.blazeEntryPointTapped(source: .productDetailPromoteButton))
                }
```

Reply:
Oh TIL! Thanks.